pretyping/pretyping.ml

```diff
@@ -705,9 +722,14 @@ struct
       (* Ne faudrait-il pas s'assurer que hyps est bien un
          sous-contexte du contexte courant, et qu'il n'y a pas de Rel "caché" *)
       let id = interp_ltac_id env id in
-      let evk =
-        try Evd.evar_key id sigma
-        with Not_found -> error_evar_not_found ?loc:locid !!env sigma id in
+      let sigma, evk =
+        try sigma, Evd.evar_key id sigma
+        with
+        | Not_found when not flags.unify_patvars ->
+            let k = Evar_kinds.(MatchingVar (FirstOrderPatVar id)) in
+            let sigma, uj_val, _ = new_typed_evar env sigma ~naming:(IntroIdentifier id) ~src:(loc,k) tycon in
+            sigma, fst (destEvar sigma uj_val)
```

Comment:
What's going on here?

Reply:
@ppedrot told me I should not have modified `pretype_patvar` without adding a flag, so I used a flag and moved the behavior to `pretype_evar` since there is no reason to use `patvar` and pattern parsing anymore.
It should do the same as before